include/DataFile.h

```diff
@@ -102,6 +102,9 @@ class LMMS_EXPORT DataFile : public QDomDocument
 
 	void cleanMetaNodes( QDomElement de );
 
+	void renameElements(QDomNodeList& elements, const QString newTagName);
+	void renameAttribute(QDomNodeList& elements, const QString oldName, const QString newName);
```

Comment:
In function declarations, top level `const` is meaningless - whether or not the variable is mutable is irrelevant when there is no function body to mutate it. Perhaps these strings were meant to be passed as references, but if not, I'd omit the `const` to reduce visual clutter. (If you choose to remove it, you can still specify it in the defintion - top level cv-qualifiers are ignored when determining the function type and matching definitions to declarations.)

Reply:
> Perhaps these strings were meant to be passed as references

As I've seen and understand it, Qt classes don't need to be passed as reference because they're essentially a wrapper over a pointer to implement copy-on-write behavior.

Reply:
That's true, but reference counting still has some overhead, and Qt itself seems to prefer to pass `QString`s by reference. In particular, `QDomElement::attribute`, `QDomElement::hasAttribute`, `QDomElement::removeAttribute`, `QDomElement::setAttribute`, and `QDomElement::setTagName`, which is where these parameters are passed on to, all take a `const QString&`.